pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -65,27 +66,32 @@ func (admitter *PodEvictionAdmitter) Admit(ar *admissionv1.AdmissionReview) *adm
 		markForEviction = true
 	}
 
-	if markForEviction && !vmi.IsMarkedForEviction() && vmi.Status.NodeName == launcher.Spec.NodeName {
+	if markForEviction && !vmi.IsMarkedForEviction() && vmi.Status.NodeName == pod.Spec.NodeName {
 		dryRun := ar.Request.DryRun != nil && *ar.Request.DryRun == true
-		err := admitter.markVMI(ar, vmi.Name, vmi.Status.NodeName, dryRun)
+		err := admitter.markVMI(vmi.Namespace, vmi.Name, vmi.Status.NodeName, dryRun)
 		if err != nil {
 			// As with the previous case, it is up to the user to issue a retry.
 			return denied(fmt.Sprintf("kubevirt failed marking the vmi for eviction: %s", err.Error()))
 		}
+
+		if !dryRun {
+			return denied(fmt.Sprintf("Eviction triggered evacuation of VMI \"%s/%s\"", vmi.Namespace, vmi.Name))
+		}
 	}
 
 	// We can let the request go through because the pod is protected by a PDB if the VMI wants to be live-migrated on
```

Comment:
Can you remove this comment? You now deny the request so the comment does not reflect reality anymore.

Reply:
This comment is still correct.
For example:
1. This is not the first time the virt-launcher pod is evicted (this happens in a loop when running `kubectl drain`)
2. A virt-launcher pod of a non migratable VMI is evicted.

Reply:
The full table is located here: https://github.com/kubevirt/kubevirt/blob/main/docs/handling-eviction-requests.md#eviction-approval-summary

kubectl output can be found [here](https://github.com/kubevirt/kubevirt/pull/11532#issuecomment-2094766310).

Reply:
Gotcha, I was missing the loop part.